src/main.rs

```diff
@@ -39,26 +74,61 @@ fn longest_common_subsequence(file1: &[String], file2: &[String]) -> Vec<Vec<usi
 
 /// Prints out the difference between the two given vectors of strings, using their previously
 /// calculated largest common subsequence grid.
-fn print_diff(subsequence_comparison_grid: &Vec<Vec<usize>>, file1: &[String], file2: &[String], file1_iter: usize, file2_iter: usize) {
+/// The following example shows how the grid is traversed, starting by the bottom-right corner:
+/// ```
+///  | |a|b|c|d     | |a|b|c|d     | |a|b|c|d     | |a|b|c|d     | |a|b|c|d     | |a|b|c|d
+///  |0|0|0|0|0     |0|0|0|0|0     |0|0|0|0|0     |0|0|0|0|0     |0|0|0|0|0     |0|X|0|0|0
+/// a|0|1|1|1|1    a|0|1|1|1|1    a|0|1|1|1|1    a|0|1|1|1|1    a|0|1|X|1|1    a|0|1|X|1|1
+/// c|0|1|1|2|2 -> c|0|1|1|2|2 -> c|0|1|1|2|2 -> c|0|1|1|X|2 -> c|0|1|1|X|2 -> c|0|1|1|X|2
+/// d|0|1|1|2|3    d|0|1|1|2|3    d|0|1|1|2|X    d|0|1|1|2|X    d|0|1|1|2|X    d|0|1|1|2|X
+/// e|0|1|1|2|3    e|0|1|1|2|X    e|0|1|1|2|X    e|0|1|1|2|X    e|0|1|1|2|X    e|0|1|1|2|X
+/// ```
+///
+/// After positioning the start, the function checks if for the max between the top and left cell,
+/// moving the iteration to the max of this two values. If both values are smaller than the value
+/// at the current cell, this means a match was found and the iteration moves diagonally; while
+/// adding the string to the largest common subsequence.
+fn print_diff(
+    subsequence_comparison_grid: &Vec<Vec<usize>>,
+    file1: &[String],
```

Comment:
I'm not 100% sure, but I suspect you could improve this interface by passing just the slices without the iter arguments. You would select the corresponding subslice when calling this function recursively, instead of just updating these iter counter